macros/src/lib.rs

```diff
@@ -790,6 +823,44 @@ fn extract_cfgs(attrs: Vec<Attribute>) -> (Vec<Attribute>, Vec<Attribute>) {
     (cfgs, not_cfgs)
 }
 
+fn check_attr_whitelist(attrs: &[Attribute]) -> Result<(), TokenStream> {
+    let whitelist = &["doc", "link_section", "cfg", "allow", "warn", "deny", "forbid", "cold"];
+    let cortex_m_rt_blacklist = &["entry", "exception", "interrupt", "pre_init"];
```

Comment:
One issue with this is that the attributes can also be used fully-qualified (`#[cortex_m_rt::exception]`) and `#[interrupt]` is reexported by PACs, so using `#[stm32l0x2::interrupt]` also works and isn't detected here.

This isn't a terribly big issue since they still fail the whitelist check, but it's something that can potentially be improved. One way to do that would be to pass the type of macro we're currently expanding to the `check_attr_whitelist` function (entry, exception, interrupt, or pre_init), and incorporate that into the error message. It could then look like:

> this attribute is not allowed on an exception handler
> this attribute is not allowed on an interrupt handler
> this attribute is not allowed on a cortex-m-rt entry point
> this attribute is not allowed on a pre-init function

These would work regardless of whether they point to a non-whitelist attribute or a cortex-m-rt one.

We can also just do this later of course. Thoughts?

Reply:
I added so error messages are better now